Doc/library/functools.rst

```diff
@@ -383,6 +383,28 @@ The :mod:`functools` module defines the following functions:
       The :func:`register` attribute supports using type annotations.
 
 
+.. class:: singledispatchmethod(func)
+
+   Return a new :class:`singledispatchmethod` descriptor which behaves
```

Comment:
While this is implemented as a class, I'd suggest borrowing the markup and introductory text from `singledispatch` and reusing an adjusted version of that text here - from an end user's perspective, the important point is that they can use `singledispatchmethod` to decorate a method definition and have it dispatch on the first non-self argument.